ocaml/idl/ocaml_backend/gen_server.ml

```diff
@@ -319,14 +319,25 @@ let gen_module api : O.Module.t =
 			"    if (try Scanf.sscanf func \"unknown-message-%s\" (fun _ -> false) with _ -> true)";
 			"    then " ^ (debug "This is not a built-in rpc \"%s\"" [ "__call" ]);
 			"    begin match __params with";
-			"    | session_id_rpc::_->";
+			"    | session_id_rpc::host_rpc::_->";
 			"      let session_id = ref_session_of_rpc session_id_rpc in";
-			"      Session_check.check false session_id;";
-			"      (* based on the pool.has_extension call *)";
+			"      let host = ref_host_of_rpc host_rpc in";
+			"      let call_string = Jsonrpc.string_of_call call in";
+ 			"      Session_check.check false session_id;";
```

Comment:
A minor point, but it would be better to do the session check as early as possible, before processing other parts of the input. (Before the `let host = ...` line.)